components/dashboard/src/settings/Notifications.tsx

```diff
@@ -13,47 +13,63 @@ import settingsMenu from "./settings-menu";
 
 export default function Notifications() {
     const { user, setUser } = useContext(UserContext);
-    const isTransactionalMail = !user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails;
-    const toggleTransactionalMail = async ()=>{
+
+    const [isTransactionalMail, _setTransactionMail] = useState(!user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails);
+    const isTransactionalMailRef = useRef(isTransactionalMail);
```

Comment:
please, just remove the `isTransactionalMailRef`

Reply:
I don't understand why it's not needed here even though it's accessed from inside an event handler but it works. Thanks. :+1: